src/components/user/actor.repository.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@nestjs/common';
+import { CachedByArg } from '@seedcompany/common';
+import { ID, Role } from '~/common';
+
+@Injectable()
+export abstract class ActorRepository {
+  @CachedByArg()
+  async getAnonymous() {
+    return await this.upsertAgent('Anonymous');
+  }
+
+  @CachedByArg()
+  async getGhost() {
```

Comment:
What is this concept of ghost, anon, etc? Not sure I follow the application here

Reply:
Does the PR description describe this with enough clarity or are you looking for more?